tests/catalog/test_sql.py

```diff
@@ -1117,17 +1117,30 @@ def test_create_namespace_with_empty_identifier(catalog: SqlCatalog, empty_names
         lazy_fixture("catalog_sqlite"),
     ],
 )
-@pytest.mark.parametrize("namespace_list", [lazy_fixture("database_list"), lazy_fixture("hierarchical_namespace_list")])
-def test_list_namespaces(catalog: SqlCatalog, namespace_list: List[str]) -> None:
+def test_list_namespaces(catalog: SqlCatalog) -> None:
+    namespace_list = ["db", "db.ns1", "db.ns1.ns2", "db.ns2", "db2", "db2.ns1", "db%"]
     for namespace in namespace_list:
         catalog.create_namespace(namespace)
-    # Test global list
+
     ns_list = catalog.list_namespaces()
-    for namespace in namespace_list:
-        assert Catalog.identifier_to_tuple(namespace) in ns_list
-        # Test individual namespace list
-        assert len(one_namespace := catalog.list_namespaces(namespace)) == 1
-        assert Catalog.identifier_to_tuple(namespace) == one_namespace[0]
+    expected_list: list[tuple[str, ...]] = [("db",), ("db2",), ("db%",)]
+    for ns in expected_list:
+        assert ns in ns_list
```

Comment:
Why not check the full list? This way we make sure that they are equal, and that the `ns_list` doesn't contain additional elements:
```suggestion
    assert ns_list == [("db",), ("db2",), ("db%",)]
```

Reply:
Good idea, thanks!
I'm using `sorted(ns_list)` to make the test more resilient. 
Or do you think we should force a specific order in `list_namespaces()`?

Reply:
Another note, the first `assert ns_list == expected_list` does not work as expected because the catalog contains other namespaces created in other tests.
```
        ns_list = catalog.list_namespaces()
>       assert sorted(ns_list) == [("db",), ("db%",), ("db2",)]
E       AssertionError: assert [('db',), ('d...t_new',), ...] == [('db',), ('db%',), ('db2',)]
E         Left contains 109 more items, first extra item: ('my_iceberg_database-alcotyqwtpiqunaddobf',)
```

Reply:
this worked for me
```
    assert sorted(catalog.list_namespaces()) == [("db",), ("db%",), ("db2",)]
    assert sorted(catalog.list_namespaces("db")) == [("db", "ns1"), ("db", "ns2")]
    assert catalog.list_namespaces("db.ns1") == [("db", "ns1", "ns2")]
    assert catalog.list_namespaces("db.ns1.ns2") == []
```